### ISSUE TITLE
Infinity constructor missing in JS and UHC backends

### ISSUE THREAD

**Issue by identicalsnowflake** [agda/agda#2218]
```
open import Agda.Builtin.Float
open import IO
open import Data.Unit using (tt ; ⊤)

postulate
  _<_ : (a b : Float) → Set

data OpenInterval (a b : Float) : Set where
  MkOpenInterval : ∀ (x : Float) → (a < x) → (x < b) → OpenInterval a b

]0,∞[ = OpenInterval 0.0 (primFloatDiv 1.0 0.0)

upper : ∀ {a b : Float} → a < b → Float
upper {a} {b} _ = b

b : ∀ (a : ]0,∞[) → Float
b a with a
... | MkOpenInterval _ _ i = upper i

main : _
main = run (return b)
```

This type checks, but fails to compile on GHC with `Not in scope: data constructor ‘Infinity’`

**Comment by UlfNorell** [comment]
I didn't touch UHC or JS backends. JS backend almost works anyway, but for some reason it loses newlines printed after printing an exotic float literal.

**Comment by asr** [comment]
Note that in the last line of the test case

```
out >  inf  -inf   nan  -nan  |  x
out >    x     -     -     -  |  Infinity
out >    -     x     -     -  |  -Infinity
out >    -     -     x     -  |  NaN
out >    -     -     -     x  |  NaN
```

`NaN` should be `-NaN` (I filed #2221 for this issue).

**Comment by UlfNorell** [comment]
That is arguable. See my comment on that issue. The key point here is that we can compile and distinguish the literals.

**Comment by asr** [comment]
> JS backend almost works anyway, but for some reason it loses newlines printed after printing an exotic float literal.

It is not related to exotic things, see #2222.

**Comment by asr** [comment]
After the fix of #2230, the issue in the test case using the UHC backend is not related to the equality but the `macro`.

**Comment by phile314** [comment]
This seems to work now on UHC/JS backends, closing.
